homeassistant/components/vesync/coordinator.py

```diff
@@ -34,4 +34,6 @@ async def _async_update_data(self) -> dict[str, Any]:
         """Fetch data from API endpoint."""
 
         # Using `update_all_devices` instead of `update` to avoid fetching device list every time.
-        return await self.hass.async_add_executor_job(self._manager.update_all_devices)
+        await self.hass.async_add_executor_job(self._manager.update_all_devices)
+        # Vesync updates energy on applicable devices every 6 hours
+        return await self.hass.async_add_executor_job(self._manager.update_energy)
```

Comment:
Put both calls in a new function and use that function as an argument for the executor job

Reply:
That has been addressed as part of this review- https://github.com/home-assistant/core/pull/134087#pullrequestreview-2527205653

Reply:
Sorry, clearly looked at an old commit. 👍